blockchain/blockchain.go

```diff
@@ -2760,10 +2765,22 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 	}
 	// Create a new context to be used in the EVM environment
 	blockContext := NewEVMBlockContext(header, bc, author)
+
+	// when execution spec test, we can insert test GasLimit to blockContext.
+	if GasLimitInExecutionSpecTest != 0 {
+		blockContext.GasLimit = GasLimitInExecutionSpecTest
+	}
 	txContext := NewEVMTxContext(msg, header, chainConfig)
 	// Create a new environment which holds all relevant information
 	// about the transaction and calling mechanisms.
 	vmenv := vm.NewEVM(blockContext, txContext, statedb, chainConfig, vmConfig)
+
+	// when execution spec test, we can insert test GasLimit to blockContext.
+	if UseKaiaCancunExtCodeHashFee && chainConfig.Rules(header.Number).IsCancun {
+		// EIP-1052 must be activated for backward compatibility on Kaia. But EIP-2929 is activated instead of it on Ethereum
+		vm.ChangeGasCostForTest(&vmenv.Config.JumpTable, vm.EXTCODEHASH, params.WarmStorageReadCostEIP2929)
+	}
```

Comment:
If EIP-1052 fails, then let it fail and skip it (unless most tests fail due to 1052). The skip list is the accurate representation of Kaia's compatibility.

Reply:
> then let it fail and skip it

Does it means `vm.ChangeGasCostForTest`?